modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2289 @@
+    }
+  }
+}
+
+void
+NSFVAction::addRhieChowUserObjects()
+{
+  const std::string u_names[3] = {"u", "v", "w"};
+  if (_porous_medium_treatment)
+  {
+    auto params = _factory.getValidParams("PINSFVRhieChowInterpolator");
+    params.set<std::vector<SubdomainName>>("block") = _blocks;
+    for (unsigned int d = 0; d < _dim; ++d)
+      params.set<VariableName>(u_names[d]) = NS::superficial_velocity_vector[d];
+
+    params.set<VariableName>("pressure") = NS::pressure;
+    params.set<MooseFunctorName>(NS::porosity) = _porosity_name;
+    unsigned short smoothing_layers = isParamValid("porosity_smoothing_layers")
+                                          ? getParam<unsigned short>("porosity_smoothing_layers")
+                                          : 0;
+    params.set<unsigned short>("smoothing_layers") = smoothing_layers;
+
+    _problem->addUserObject("PINSFVRhieChowInterpolator", "pins_rhie_chow_interpolator", params);
+  }
+  else
+  {
+    auto params = _factory.getValidParams("INSFVRhieChowInterpolator");
+    params.set<std::vector<SubdomainName>>("block") = _blocks;
+    for (unsigned int d = 0; d < _dim; ++d)
+      params.set<VariableName>(u_names[d]) = NS::velocity_vector[d];
+
+    params.set<VariableName>("pressure") = NS::pressure;
+
+    _problem->addUserObject("INSFVRhieChowInterpolator", "ins_rhie_chow_interpolator", params);
+  }
+}
+
+void
+NSFVAction::addINSInitialConditions()
+{
+  InputParameters params = _factory.getValidParams("ConstantIC");
+  auto vvalue = getParam<RealVectorValue>("initial_velocity");
+
+  if (_porous_medium_treatment)
+    for (unsigned int d = 0; d < _dim; ++d)
+    {
+      params.set<VariableName>("variable") = NS::superficial_velocity_vector[d];
+      params.set<Real>("value") = vvalue(d);
+
+      _problem->addInitialCondition(
+          "ConstantIC", NS::superficial_velocity_vector[d] + "_ic", params);
+    }
+  else
+    for (unsigned int d = 0; d < _dim; ++d)
+    {
+      params.set<VariableName>("variable") = NS::velocity_vector[d];
+      params.set<Real>("value") = vvalue(d);
+
+      _problem->addInitialCondition("ConstantIC", NS::velocity_vector[d] + "_ic", params);
+    }
+
+  params.set<VariableName>("variable") = NS::pressure;
+  params.set<Real>("value") = getParam<Real>("initial_pressure");
+
+  _problem->addInitialCondition("ConstantIC", "pressure_ic", params);
+
+  if (_has_energy_equation)
+  {
+    params.set<VariableName>("variable") = NS::T_fluid;
+    params.set<Real>("value") = getParam<Real>("initial_temperature");
+
+    _problem->addInitialCondition("ConstantIC", NS::T_fluid + "_ic", params);
+  }
+
+  if (_passive_scalar_names.size())
+  {
+    for (unsigned int name_i = 0; name_i < _passive_scalar_names.size(); ++name_i)
+    {
+      bool initialize_me = true;
+      if (_create_scalar_variable.size())
+        if (!_create_scalar_variable[name_i])
+          initialize_me = false;
```

Comment:
but user could be passing initalization conditions even though he defined the variables outside?

Reply:
This changed, we only read initial conditions for the internal variables.